skbio/io/_base.py

```diff
@@ -185,3 +186,25 @@ def _format_fasta_like_records(generator, id_whitespace_replacement,
                 "scores associated with it." % cardinal_to_ordinal(idx + 1))
 
         yield header, seq.sequence, seq.quality
+
+
+def _line_generator(fh, skip_blanks=False):
+    for line in fh:
+        line = line.strip()
+        if line or not skip_blanks:
+            yield line
+
+
+def _too_many_blanks(fh, max_blanks):
+    count = 0
+    too_many = False
+    for line in _line_generator(fh):
+        if line:
+            break
+        else:
+            count += 1
+            if count > max_blanks:
+                too_many = True
+                break
+    fh.seek(0)
```

Comment:
The problem was when the file is valid, this function will consume the
first valid line then the file is sniffed incorrectly. There may well be a
way around this. I'd prefer to get rid of the seek.
On May 1, 2015 10:33 AM, "Jai Ram Rideout" <notifications@github.com> wrote:

> In skbio/io/_base.py
> <https://github.com/biocore/scikit-bio/pull/893#discussion_r29517672>:
>
> > +        if line or not skip_blanks:
> > +            yield line
> > +
> > +
> > +def _too_many_blanks(fh, max_blanks):
> > +    count = 0
> > +    too_many = False
> > +    for line in _line_generator(fh):
> > +        if line:
> > +            break
> > +        else:
> > +            count += 1
> > +            if count > max_blanks:
> > +                too_many = True
> > +                break
> > +    fh.seek(0)
>
> Sweet, this looks good! Minor comment: is this seek necessary? If there
> aren't too many blank lines at the beginning, I think the sniffers can just
> keep reading from the file handle's current position.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/893/files#r29517672>.
>